microsoft/testsuites/vm_extensions/azsecpack.py

```diff
@@ -0,0 +1,287 @@
+    simple_requirement,
+)
+from lisa.operating_system import BSD, CBLMariner, Posix
+from lisa.sut_orchestrator.azure.common import (
+    add_tag_for_vm,
+    add_user_assign_identity,
+    get_node_context,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.sut_orchestrator.azure.tools import Azsecd
+from lisa.testsuite import TestResult
+from lisa.tools import Cat, Curl, Journalctl, Service
+from lisa.util import UnsupportedDistroException
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="""
+    BVT for Azure Security Pack.
+    Azure Security Pack includes core security features that provide security logging
+    and monitoring coverage for the service.
+
+    This test suite validate if Azure security pack can be installed, uninstalled
+    successfully, and check if the autoconfig is configured successfully.
+
+    This test requires your subscription is within AutoConfig scope. It manually enables
+    the AzSecPack AutoConfig on the Linux VM. The steps are:
+    1. Add resource tag for AzSecPack
+    2. Create an assign user assigned managed identity AzSecPack to the VM
+    3. Add Azure VM extensions for AMA and ASA to the VM
+
+    If the subscription is within AutoConfig scope, the AutoCoonfig onboarding method
+    is recommended. It needs adding resoure tag for AzSecPack, creating and assigning
+    UserAssigned Managed Identity AzSecPack AutoConfig to the ARM resources.
+    """,
+)
+class AzSecPack(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Verify whether Azure security pack can be installed, uninstalled
+            successfully, and check if the autoconfig is configured successfully.
+        """,
+        priority=0,
+        use_new_environment=True,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension], unsupported_os=[BSD]
+        ),
+    )
+    def verify_azsecpack(self, node: Node, log: Logger, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+        rm_client = platform._rm_client
+        assert rm_client
+        msi_client = platform._msi_client
+        assert msi_client
+
+        node_context = get_node_context(node)
+        resource_group_name = node_context.resource_group_name
+        location = node_context.location
+        vm_name = node_context.vm_name
+
+        # Add resource tag for AzSecPack
+        tag = {"azsecpack": "nonprod"}
+        add_tag_for_vm(platform, resource_group_name, vm_name, tag, log)
+
+        # Create an AzSecPackAutoConfigRG resource group with the specific location
+        # For Public cloud, the specific location is "eastus".
+        autoconfig_rg_name = "AzSecPackAutoConfigRG"
```

Comment:
Is the name hard coded? Can it be put in shared resource group?

Reply:
Yes, this name is hard-coded. We can't use other names.

Reply:
Please update comments to explain the name won't be changed.

Reply:
Done